server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java

```diff
@@ -18,10 +18,66 @@
  ****************************************************************/
 package org.apache.james.util;
 
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Spliterator;
+import java.util.stream.Stream;
+
+import reactor.core.publisher.Flux;
 import reactor.core.publisher.Mono;
 
 public class ReactorUtils {
     public static <T> Mono<T> executeAndEmpty(Runnable runnable) {
         return Mono.fromRunnable(runnable).then(Mono.empty());
     }
+
+    public static InputStream toInputStream(Flux<byte[]> byteArrays) {
+        return new StreamInputStream(byteArrays.toStream());
+    }
+
+    private static  class StreamInputStream extends InputStream {
```

Comment:
The code style of this class needs some enhancements IMO

Reply:
be more specific please

Reply:
null, magic constants (-1), the read method is a bit too long and a bit too complex (I need to think hard to understand what's going on), non explicit if conditions...

Mutability dark magic needs some explanations (extracted methods), in order to make this readable.

Let me give it a try

Reply:
I did what I could... Mutability is hard. Feel free to modify my fixup at will...